[ENH] BEP031 - New entity: sample and samples.tsv file (#812)
bids-standard/bids-specification: src/03-modality-agnostic-files.md

}
```

## Samples file

Template:

```Text
samples.tsv
samples.json
```

> The purpose of this RECOMMENDED file is to describe properties of samples

Reviewer: should this be recommended or required if you have a sample entity. i don't think i would want people to use the sample entity to run disambiguation routes, and we would want to know the type of sample.

Author: I addressed this in another similar question [here](https://github.com/bids-standard/bids-specification/pull/812#discussion_r646894795).

> The purpose of this RECOMMENDED file is to describe properties of samples
> when the `sample-<label>` key/value pair is used in filenames.

Reviewer 2: I agree with @satra that this should be REQUIRED if `sample` is present. Something like the following?

```suggestion
The purpose of this file is to describe properties of samples, indicated by the `sample` entity.
This file is REQUIRED if `sample-<label>` is present in any file name within the dataset.
```

Author: I went with “recommended” because I think all the other `tsv` files in the specs are either recommended or optional, even `participants.tsv`. But I agree with @satra that the `sample_type` column is reason enough to make the file REQUIRED, I was just wondering if that makes sense with the rest of the spec? If so, I agree with the suggested changes.

Reviewer 2: participants.tsv is not REQUIRED because there is no required metadata in it. if `sample_type` must exist for a dataset to be interpretable, then this should be REQUIRED.

If this file exists, it MUST contain the three following columns:

-   `sample_id`: MUST consist of `sample-<label>` values identifying one row
    for each sample

-   `participant_id`: MUST consist of `sub-<label>`

-   `sample_type`: MUST consist of sample type values (for example: tissue,

>     primary cell or cell-free sample) from [ENCODE Biosample Type](https://www.encodeproject.org/profiles/biosample_type)

Reviewer 2: This may be overly restrictive. We do have the concept of TermURL, that allows researchers to associate a column with any ontology.

No strong opinion, here, though. If we want to require this TermURL for now, it can be loosened in the future to permit other ontologies.

Author: For Apps down the line, it seems easier to restrict the possible values.

I have question about validation though:
1. If the value MUST be from a specific ontology, does it mean that we have to plan a validation for it?
2. And if the value is REQUIRED but the ontology is not, what would we validate?

Author: The current set of Encode Biosample is fairly small:
`"cell line", "in vitro differentiated cells", "primary cell", "cell-free sample", "cloning host", "tissue", "whole organisms", "organoid", "technical sample"`. I could list them all in the description as accepted values if we go with a fixed ontology.

As for TermURL, I understand it the same way as @satra, as a link to the definition of what `sample_type` means rather than the link to the ontology itself. If we want to point to one of the Ontobee definition, I would go with `Biospecimen` (http://www.ontobee.org/ontology/NCIT?iri=http://purl.obolibrary.org/obo/NCIT_C70699) rather than `TissueSample` (http://purl.obolibrary.org/obo/NCIT_C19697). However, it feels strange to have a `TermURL` linking to a different ontology than the one we wish to use for the values.

Reviewer 2: You're right about TermURL. Apologies for the confusion. I would probably go with a fixed set, and we can say that these are taken from Encode Biosample.

Other optional columns MAY be used to describe the samples.
Each sample MUST be described by one and only one row.

Commonly used *optional* columns in `samples.tsv` files are `pathology` and
`derived_from`. We RECOMMEND to make use of these columns, and in case that
you do use them, we RECOMMEND to use the following values for them:

-   `pathology`: string value describing the pathology of the sample or type of control.
    When different from `healthy`, pathology SHOULD be specified in `samples.tsv`.
    The pathology MAY instead be specified in [Sessions files](06-longitudinal-and-multi-site-studies.md#sessions-file)
    in case it changes over time.

-   `derived_from`: `sample-<label>` key/value pair from which a sample is derived from,
    for example a slice of tissue (`sample-02`) derived from a block of tissue (`sample-01`),

>     as illustrated in the example below.

Reviewer: this makes me think that a samples.tsv could refer to entities that are not in the filenames. which would be fine. but we may want to illustrate or clarify.

Author: I’m sorry, I don’t understand what you mean, could you give me an example. Thanks!

Reviewer: imagine you submit data from a slab (slab-01), which came from a block (sample-01). both are samples, and one may want to provide properties for both, even though the data itself is about the slab. so:

```
sample-01, tissue, sub-01
slab-01, slice, sample-01
```

Author: In the current proposal, if `slab-01` and `sample-01` are both samples, they are required to be identified with a `sample-<label>`. This was the idea behind `sample` to avoid adding several entities to the specs like `block`, `slice`, `slab`, etc. So any of these should be identified as a `sample`. From your example, that would give something like this, where the `id` is any alphanumeric string:

|sample_id|participant_id   |sample_type|derived_from|
|---------|-----------------|-----------|------------|
|sample-01|sub-01         |tissue     |n/a         |
|sample-slab01|sub-01         |tissue     |sample-01|

In the first line: because every sample is always associated with a subject (`ex: sub-01_sample-01_SEM.ome.tif`), `sub-01` goes in `participant_id` and not in `derived_from`.
In the second line: `slice` is not one of the `sample_type` accepted in ENCODE biosample, they would both be `tissue` in this case.

Let me know if I interpreted correctly your comment. Thanks!

Reviewer: thank you @mariehbourget - this looks great.

`samples.tsv` example:

```Text
sample_id participant_id sample_type derived_from
sample-01 sub-01 tissue n/a
sample-02 sub-01 tissue sample-01
sample-03 sub-01 tissue sample-01
sample-04 sub-02 tissue n/a
sample-05 sub-02 tissue n/a
```

It is RECOMMENDED to accompany each `samples.tsv` file with a sidecar
`samples.json` file to describe the TSV column names and properties of their values
(see also the [section on tabular files](02-common-principles.md#tabular-files)).

`samples.json` example:

```JSON
{
    "sample_type": {

>         "Description": "type of sample from ENCODE Biosample Type (https://www.encodeproject.org/profiles/biosample_type)",

Reviewer 2: ```suggestion
        "Description": "type of sample from ENCODE Biosample Type",
        "TermURL": "https://www.encodeproject.org/profiles/biosample_type"
```

    },
    "derived_from": {

>         "Description": "sample_id from which the sample is derived from",

Reviewer 2: ```suggestion
        "Description": "sample_id from which the sample is derived"
```
